### ISSUE TITLE
[$2000] No error displayed using invalid set password link from email - reported by @K4tsuki

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#8124]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Sign out
2. Open https://new.expensify.com/setpassword/1138003/ULXLJCUMX I got from email
3. verify application doesn't show invalid link error in setpassword page and proceed to set password.
4. After set password, the browser tab close abruptly

## Expected Result:
Setpassword page must display invalid link or expired link and prevent set password action

## Actual Result:
Setpassword page doesn't display error message, then proceed with setpassword and then close the tab abruptly

## Workaround:
unknown 

## Platform:

Where is this issue occurring?

- Web
- Mobile Web

**Version Number:** 1.1.41-2
**Reproducible in staging?:** Y 
**Reproducible in production?:** Y 
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**  

https://user-images.githubusercontent.com/43995119/158105579-94db636b-9089-4a1b-a8f9-b2810f829b98.mp4

**Issue reported by:** @K4tsuki
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1645845370601599

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @conorpendergrast (`AutoAssignerTriage`), see https://stackoverflow.com/c/expensify/questions/4749 for more details.

**Comment by conorpendergrast** [comment]
Interesting! Reproduced now

**Comment by MelvinBot** [comment]
Triggered auto assignment to @NikkiWines (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by NikkiWines** [comment]
Oh interesting, I actually experienced this yesterday while [testing out a different bug](https://expensify.slack.com/archives/C01GTK53T8Q/p1647281953734039?thread_ts=1647270733.031429&cid=C01GTK53T8Q)

https://user-images.githubusercontent.com/16747822/158443313-1afc964c-ba2e-4698-a8ea-d9908e7fc7fd.mp4


Either way, this looks good for UpWork I think since I doubt we've changed anything in the backend that would result in this and the issue isn't reproducible on expensify.com

**Comment by MelvinBot** [comment]
Triggered auto assignment to @Christinadobrzyn (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by K4tsuki** [comment]
@NikkiWines There is no assigneed people for this issue.

#### proposal:

Need back end change:
Currently we cannot be sure if validation link is expired or not until user set password:

https://github.com/Expensify/App/blob/8ea8177869f22f7aa750470d1d0b54d9e379cd7a/src/pages/SetPasswordPage.js#L107

For sign up user, `setpassword link` / validation link is checked by sending:

https://github.com/Expensify/App/blob/8ea8177869f22f7aa750470d1d0b54d9e379cd7a/src/pages/SetPasswordPage.js#L98 

`Session.validateEmail`. The link can be used only once, so if the `Session.validateEmail` is being sent again, the json code reply is not `200` and response message is like `already validated you can login...` . 

<br />

Instead we can provide information/error when user visit the link and display message there. If validation link is expired we display the message and show `resend validation link` button instead of set password text input.

My suggestion is to rovide api for retrieving validation link details about:
- Is it from forgot link password or from sign up user. Forgot password link doesn't need sign up authToken to setpassword.
- Validation link status (is it expired or not) 
- Email address/login (who is the validation link for) (related to this [PR](https://github.com/Expensify/App/pull/7930) that require login email to be displayed when user setpassword)) that will display email/login in the setpassword page 

Also related to this [issue](https://github.com/Expensify/App/issues/8130).

**Comment by NikkiWines** [comment]
Oh oops, thanks for the heads up @K4tsuki - I accidentally unassigned @Christinadobrzyn

**Comment by Christinadobrzyn** [comment]
Posted job in Upwork: 

Internal posting - https://www.upwork.com/ab/applicants/1503913194252062720/job-details
External posting - https://www.upwork.com/jobs/~017bbe06765188f815

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @rushatgabhane (`Exported`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @deetergp (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by mallenexpensify** [comment]
Doubled price to $500, re-added `Help Wanted` label
https://www.upwork.com/jobs/~017bbe06765188f815

**Comment by mallenexpensify** [comment]
@rushatgabhane  can you please review this proposal above
https://github.com/Expensify/App/issues/8124#issuecomment-1068608367

If the proposal is complete-enough and is one you'd recommend accepting, I want to compensate the contributor who we're not working with anymore

**Comment by mountiny** [comment]
Just tested https://github.com/Expensify/Web-Expensify/pull/33368

**Comment by rushatgabhane** [comment]
> Just tested https://github.com/Expensify/Web-Expensify/pull/33368

@mountiny any interesting finds?

**Comment by mountiny** [comment]
@rushatgabhane yeah sorry :D the Help Wanted label has not been removed so 🎉 fixed

**Comment by rushatgabhane** [comment]
> fixed

The issue is fixed? If so, perfect! I'll unassign myself

**Comment by MelvinBot** [comment]
📣 @rushatgabhane You have been assigned to this job by @mountiny!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md) 📖

**Comment by mountiny** [comment]
@rushatgabhane apologies for not being clearer 😂 this issue is not fixed, i meant the issue (which you unfortunately cannot see) related to the contributor management has been fixed. You could see in last couple of days that you were assigned as C+ but the bot thought you are contributor assigned to fix the issue.

I assigned you back as C+

**Comment by mallenexpensify** [comment]
Price doubled to $1000 
https://www.upwork.com/jobs/~017bbe06765188f815

**Comment by rushatgabhane** [comment]
I don't think we need any backend changes for this issue.

@mallenexpensify we won't be going forward with @K4tsuki's proposal.

### Success
<img width="650" alt="Screen Shot 2022-03-29 at 11 12 57 PM" src="https://user-images.githubusercontent.com/29673073/160674578-bf65a51d-7fe6-49c6-aaf1-7fb723e85905.png">


### Fail
<img width="917" alt="Screen Shot 2022-03-29 at 11 23 06 PM" src="https://user-images.githubusercontent.com/29673073/160674589-d3deb7fe-b3e3-4f15-b866-e46781ad7055.png">

<img width="690" alt="Screen Shot 2022-03-29 at 11 25 47 PM" src="https://user-images.githubusercontent.com/29673073/160674988-b089ce35-87c7-4a01-a77f-5dffd42ac35a.png">

**Comment by mallenexpensify** [comment]
I should double this to $2000, right @rushatgabhane ?

**Comment by rushatgabhane** [comment]
@mallenexpensify yes, we don't have any good proposals yet.

**Comment by mallenexpensify** [comment]
Thanks @rushatgabhane 
Doubled price to $2000
https://www.upwork.com/jobs/~017bbe06765188f815

**Comment by thesahindia** [comment]
Is this still reproducible? These are the actions I performed -
1. Added my email at login page and in password page I chose "forgot"
2. Pressed "resend link"
3. Copied the url of first link
4. Pasted the url in the browser and added new password
5.  Saw this -
<img width="379" alt="Screenshot 2022-04-13 at 7 26 29 PM" src="https://user-images.githubusercontent.com/83179295/163197708-e2cd54d3-73cf-46a1-9a90-a383cdf99dbe.png">

Also there's a minor bug here, the message says "validate code" instead of "validation link" should I report it on slack?

**Comment by rushatgabhane** [comment]
@mallenexpensify we should create a new issue right?

**Comment by mallenexpensify** [comment]
If it's not reproducible, we should close this.  @rushatgabhane , it doesn't look like anyone is due  compensation, can you confirm?

@thesahindia can you report the bug you found in #expensify-open-source using the bug template? If you think it's helpful or relevant to link back to this issue, go for it

**Comment by rushatgabhane** [comment]
Let's close this then.
